prow/plugins/plugins.go

```diff
@@ -221,6 +221,7 @@ type Configuration struct {
 	Size                 *Size                  `json:"size,omitempty"`
 	Triggers             []Trigger              `json:"triggers,omitempty"`
 	Welcome              []Welcome              `json:"welcome,omitempty"`
+	ProjectQueries       []ProjectQuery         `json:"project_queries,omitempty"`
```

Comment:
I would not necessarily call these `queries` as those have a specific technical definition when in the context of the GitHub API

Reply:
Instead of having a loose list of queries, let's key these either on org or org/repo -- look at the BranchProtection configuration. That sort of tree structure with overrides is more natural for configuring this sort of behavior and from our experience easier for users to interface with.

Reply:
Would that be keyed to the org/repo of the project? So for example, to add to the KEP Tracking project you would add the following to `plugins.yaml`:
https://github.com/kubernetes-sigs/architecture-tracking/projects/2
```
project_queries:
  orgs:
    kubernetes-sigs:
      repos:
        architecture-tracking:
          projects:
            KEP Tracking:
              columns:
                Backlog:
                  state: open
                  org: kubernetes
                  labels:
                    - kind/kep
```
Or possibly something a bit less verbose, but along those lines?

Reply:
Yeah, something like that. If there's no sense in having projects that span multiple repos we can just key on `org/repo`

Reply:
I'd also suggest something like:

```yaml
projects:
- name: "KEP Tracking"
```

Instead of 

```yaml
projects:
  KEP Tracking:
```

For ease of use once it's ingested into Go structs

Reply:
On the other hand using the `projects: {"KEP Tracking": ...}` pattern makes it easier to prevent duplicate names.

Reply:
I'm now thinking of structuring it like this:

```
project_queries:
  org/repo:
    kubernetes-sigs/architecture-tracking:
      projects:
      - name: "KEP Tracking"
        columns:
        - name: "Backlog"
           state: open
           org: kubernetes
           labels:
           - kind/kep
```

This groups org and repo so the yaml gets less indented and should be easier to read and recognise org/repos. Org level projects can also be referenced by not specifying a `/`.

Each org/repo then has a list of projects so that easily translates to a `map[string][]Project`. Each project has a name and list of columns, and each column has a name and set of conditions so these easily translate to a `project` and `column` struct.

As a side note I plan to change the structure of the `column` part so `org` is `orgs` and a list of orgs to add PRs from can be given. Other terms could also be added to this as required for more flexible matching.